resources/services/frontdoor/front_doors_mock_util_test.go

```diff
@@ -0,0 +1,289 @@
+package frontdoor
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/frontdoor/mgmt/2020-11-01/frontdoor"
+	"github.com/cloudquery/faker/v3"
+)
+
+func fakeBasicRouteConfiguration(t *testing.T) frontdoor.BasicRouteConfiguration {
+	switch rand.Int() % 3 {
```

Comment:
I'm a bit wary of using the `rand` package directly in tests - we don't know if something else might be changing the seed and therefore causing the test to run non-deterministically. I see two options:
 1. Instantiate a new `Rand` object and use it throughout, so we know the seed is fixed, e.g. `r := rand.New(rand.NewSource(1))`
 2. Keep track of a counter to do the modulo operation on, and increment it each time this function is called.
 
My preference would lean slightly towards #1, but either one is fine imo, as long as there's no chance of the test behavior changing randomly either now or in the future if the seed gets changed somewhere else.

Reply:
Good catch. Agreed. better not to use any random stuff in tests